src/components/WeatherForm.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import { fetchWeather } from '../utils'
+import WeatherResults from './WeatherResults'
+
+class WeatherForm extends Component{
+  state = {
```

Comment:
Have you used React hooks for [state management](https://reactjs.org/docs/hooks-intro.html) like this before? Any particular opinions?

Reply:
Yes, I have. As well as Redux for dealing with large amount of data. 
The benefits of using Hooks include: Easier to scale up.  It supports multiple life cycle hooks of the same type. Less probability of data leakage, in case the component unmount before promise is resolved.
The reason why I did not add it here is because I wanted to be able to show how I would deal with state. Since sometimes, part of the job is to rewrite class components as functional components and use Hooks or Redux.
I am happy to refactor though.